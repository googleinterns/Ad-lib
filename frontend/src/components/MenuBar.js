import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import HelpIcon from '@material-ui/icons/Help';

/**
 * Establishes style to use on rendering menu bar component
 */
const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

/**
 * Create menu bar component with title and help icon
 * @return {MenuBar} MenuBar component
 */
export default function MenuAppBar() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Ad-lib
          </Typography>
          {
            <div>
              <IconButton
                aria-label="help-icon"
                aria-controls="help-appbar"
                color="inherit"
              >
                <HelpIcon />
              </IconButton>
            </div>
          }
        </Toolbar>
      </AppBar>
    </div>
  );
}
