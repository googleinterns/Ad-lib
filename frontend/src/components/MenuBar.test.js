import React from 'react';
import MenuBar from './MenuBar';
import renderer from 'react-test-renderer';

describe('Menu Bar', () => {
  it('should be defined', () => {
    expect(MenuBar).toBeDefined();
  });
  test('should render correctly', () => {
    const tree = renderer.create(<MenuBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
