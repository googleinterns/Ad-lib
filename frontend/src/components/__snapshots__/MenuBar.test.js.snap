// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Bar should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Ad-lib
      </h6>
      <div>
        <button
          aria-controls="help-appbar"
          aria-label="help-icon"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>
</div>
`;
